src/Boards.h

```diff
@@ -123,7 +123,7 @@ writePort(port, value, bitmask):  Write an 8 bit port.
 // #define TOTAL_PINS              21 //defined in spark_wiring.h
 // #define TOTAL_ANALOG_PINS       8 //defined in spark_wiring.h
 #define VERSION_BLINK_PIN       7
-#define IS_PIN_DIGITAL(p)       ((p) >= 0 && (p) < TOTAL_PINS && (p) != 8 && (p) != 9  && (p) != 20)
+#define IS_PIN_DIGITAL(p)       ((p) >= 0 && (p) < TOTAL_PINS && (p) != 8 && (p) != 9  && (p) != 20 && (p) != 21 && (p) != 22 && (p) != 23)
 #define IS_PIN_ANALOG(p)        ((p) >= FIRST_ANALOG_PIN && (p) < (FIRST_ANALOG_PIN+TOTAL_ANALOG_PINS))
 #define IS_PIN_PWM(p)           ((p) == 0 || (p) == 1 || (p) == 10 || (p) == 11 || (p) == 14 || (p) == 15 || (p) == 16 || (p) == 17)
```

Comment:
You may be able to use the macro [digitalPinHasPWM(p)](https://github.com/spark/firmware/blob/develop/wiring/inc/spark_wiring_arduino.h#L125) here.